langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/ToExpString.java

```diff
@@ -0,0 +1,120 @@
+        }
+
+        long noOfFractionDigits;
+
+        // If fractionalDigits is `()`, use the minimum number of digits required to accurately represent the value.
+        double xAbsValue = Math.abs(x);
+        if (fractionDigits == null) {
+            if (xAbsValue == 0) {
+                noOfFractionDigits = BigDecimal.valueOf(xAbsValue).scale();
+            } else {
+                int integerPart = (int) (Math.log10(xAbsValue));
+                noOfFractionDigits = BigDecimal.valueOf(xAbsValue / Math.pow(10, integerPart)).scale();
+            }
+        } else {
+            noOfFractionDigits = (long) fractionDigits;
+        }
+
+        // Panic if noOfFractionDigits < 0.
+        if (noOfFractionDigits < 0) {
+            throw ErrorCreator.createError(getModulePrefixedReason(FLOAT_LANG_LIB,
+                    BallerinaErrorReasons.INVALID_FRACTION_DIGITS_ERROR),
+                    BLangExceptionHelper.getErrorDetails(RuntimeErrors.INVALID_FRACTION_DIGITS));
+        }
+
+        // Handle very large int values since they might cause overflows.
+        if (noOfFractionDigits > Integer.MAX_VALUE ||
+                (noOfFractionDigits * Long.bitCount(noOfFractionDigits) > Integer.MAX_VALUE)) {
+            // Maximum value a double can represent is around 1.7*10^308.
+            noOfFractionDigits = 308;
+        }
+
+        int exponent = (int) noOfFractionDigits;
+
+        int tens = 0;
+        if (xAbsValue != 0 && xAbsValue < 1) {
+            double multipliedValue = xAbsValue;
+
+            while (multipliedValue < 1) {
+                multipliedValue = xAbsValue * Math.pow(10, tens + 1);
+                tens++;
+            }
+        }
+
+        BigDecimal numberBigDecimal = new BigDecimal(x);
+        if (xAbsValue != 0 && xAbsValue < 1) {
+            numberBigDecimal  = numberBigDecimal.setScale(exponent + tens, RoundingMode.HALF_EVEN);
+
+        } else {
+            numberBigDecimal  = numberBigDecimal.setScale(exponent, RoundingMode.HALF_EVEN);
+        }
+
+        String power = "0".repeat(exponent);
+
+        DecimalFormat decimalFormat = new DecimalFormat("0." + power + "E0");
+        decimalFormat.setRoundingMode(RoundingMode.HALF_EVEN);
+        String res = decimalFormat.format(numberBigDecimal);
+        int indexOfExp = res.lastIndexOf("E");
+        String firstSection = res.substring(0, indexOfExp);
+        int idxOfDecimalPoint = firstSection.lastIndexOf(".");
+        if (idxOfDecimalPoint == firstSection.length() - 1) {
+            firstSection = res.substring(0, idxOfDecimalPoint);
+        }
+        String secondSection = res.substring(indexOfExp + 1);
+        int p = Integer.parseInt(secondSection);
+        if (p >= 0) {
+            secondSection = "e+" + secondSection;
+        } else {
+            secondSection = "e" + secondSection;
+        }
+
+        return StringUtils.fromString(firstSection + secondSection);
```

Comment:
Can we not generate the pattern with `#` and simply generate the ExpString? May be we can address it in a separate PR

Reply:
Will address this separately if the PR does not need any further changes.

Reply:
Refer the java.text.DecimalFormat when you do

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/35757